source/dyaml/serializer.d

```diff
@@ -66,12 +66,12 @@ struct Serializer
          *          YAMLVersion   = YAML version string.
          *          tagDirectives = Tag directives to emit.
          */
-        this(ref Emitter emitter, Resolver resolver, Encoding encoding,
+        this(Emitter* emitter, Resolver resolver, Encoding encoding,
              const Flag!"explicitStart" explicitStart,
              const Flag!"explicitEnd" explicitEnd, string YAMLVersion,
-             TagDirective[] tagDirectives) @trusted
+             TagDirective[] tagDirectives) @safe
         {
-            emitter_       = &emitter;
+            emitter_       = emitter;
```

Comment:
it's indeed `@safe` but now there's a full copy construction of the ref parameter. This can be a problem if an Emitter stores reference to his own `this`.

Reply:
there is no ref parameter anymore - it is a pointer now.

Reply:
yeah, sorry. I mismatched red and green